### ISSUE TITLE
Fix autofix running on files containing scoped disable comments

### ISSUE THREAD

**Issue by amyspark** [stylelint/stylelint#4704]
> Clearly describe the bug

`npx stylelint --fix` corrects violations of the rule `color-hex-length` even after disabling it with `// stylelint-disable color-hex-length`.

> Which rule, if any, is the bug related to?

`color-hex-length`

> What code is needed to reproduce the bug?

This is part of a bigger file located (in my system) at `_sass/layout/syntax-lighlighting.scss`.

```scss
// https://github.com/richleland/pygments-css/blob/master/vs.css
// stylelint-disable color-hex-length, color-named, font-weight-notation, selector-class-pattern
.highlight .hll { background-color: #ffffcc; }
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-twbs-bootstrap/scss"
  ],
  "rules": {
    "string-quotes": null,
    "indentation": null,
    "value-list-comma-newline-after": null,
    "value-list-comma-space-after": null,
    "function-calc-no-unspaced-operator": null,
    "scss/operator-no-unspaced": null,
    "declaration-no-important": null,
    "declaration-colon-newline-after": null,
    "order/properties-order": null,
    "scss/at-function-named-arguments": null
  }
}
```

> Which version of stylelint are you using?

13.3.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`npx stylelint _sass/**/*.scss --fix`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

No corrections made.

> What actually happened (e.g. what warnings or errors did you get)?

`npx stylelint _sass/**/*.scss` yields nothing, as expected. But `npx stylelint _sass/**/*.scss --fix` ignores the disabling comment and yields:
```scss
// https://github.com/richleland/pygments-css/blob/master/vs.css
// stylelint-disable color-hex-length, color-named, font-weight-notation, selector-class-pattern
.highlight .hll { background-color: #ffc; }
```

**Comment by jeddy3** [comment]
@amyspark Thanks for the clear report and for using the template.

I can reproduce this locally using SCSS and vanilla CSS.

As of [13.2.0](https://stylelint.io/changelog#1320), stylelint should not attempt to autofix sources containing disable comments (as a temporary measure to make autofix safer to use until https://github.com/stylelint/stylelint/issues/2643 is resolved). This temporary measure works for unscoped disabled comments, i.e. `/* stylelint-disable */` and `// stylelint-disable`, but not scoped ones, e.g. `/* stylelint-disable color-hex-length */`. I'll look into why that is.

**Comment by jeddy3** [comment]
Released in `13.3.3`.
